packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -0,0 +1,236 @@
+import * as React from 'react';
+
+import { Button, Flex, VisuallyHidden } from '@strapi/design-system';
+import { Menu } from '@strapi/design-system/v2';
+import { More } from '@strapi/icons';
+import { useIntl } from 'react-intl';
+import { useNavigate } from 'react-router-dom';
+
+import { DocumentActionComponent } from '../../../../core/apis/content-manager';
+import { useForm } from '../../../components/Form';
+import { useDocumentRBAC } from '../../../features/DocumentRBAC';
+import { useDocumentActions } from '../../../hooks/useDocumentActions';
+
+/* -------------------------------------------------------------------------------------------------
+ * Types
+ * -----------------------------------------------------------------------------------------------*/
+
+interface DocumentActionDescription {
+  label: string;
+  onClick: React.MouseEventHandler<HTMLButtonElement>;
+  icon?: React.ReactNode;
+  /**
+   * @default false
+   */
+  disabled?: boolean;
+  /**
+   * @default 'panel'
+   * @description Where the action should be rendered.
+   */
+  position?: 'panel' | 'header';
+  dialog?: DialogOptions | NotificationOptions | ModalOptions;
+}
+
+interface DialogOptions {
+  type: 'dialog';
+  title: string;
+  content?: React.ReactNode;
+  onConfirm?: () => void;
+  onCancel?: () => void;
+}
+
+interface NotificationOptions {
+  type: 'notifcation';
+  title: string;
+  link?: {
+    label: string;
+    url: string;
+    target?: string;
+  };
+  content?: string;
+  onClose?: () => void;
+  status?: 'info' | 'warning' | 'softWarning' | 'success';
+  timeout?: number;
+}
+
+interface ModalOptions {
+  type: 'modal';
+  title: string;
+  content: React.ReactNode;
+  footer: React.ReactNode;
+  onClose?: () => void;
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * DocumentActions
+ * -----------------------------------------------------------------------------------------------*/
+
+interface DocumentActionsProps {
+  actions: DocumentActionDescription[];
+}
+
+const DocumentActions = ({ actions }: DocumentActionsProps) => {
+  const { formatMessage } = useIntl();
+
+  const [primaryAction, secondaryAction, ...restActions] = actions.filter(
+    (action) => action.position !== 'header'
+  );
+
+  if (!primaryAction) {
+    return null;
+  }
+
+  const isMenuDisabled = restActions.every((action) => action.disabled);
+
+  return (
+    <Flex direction="column" gap={2} alignItems="stretch" width="100%">
+      <Flex gap={2}>
+        <Button
+          flex={1}
+          startIcon={primaryAction.icon}
+          disabled={primaryAction.disabled}
+          onClick={primaryAction.onClick}
+          justifyContent="center"
+        >
+          {primaryAction.label}
+        </Button>
+        {restActions.length > 0 ? (
```

Comment:
restActions are not actually implemented yet right? I don't see the value being used anywhere